lib/websocket_proxy.rb

```diff
@@ -15,7 +15,7 @@ def initialize(env, console)
       # Hijack the socket from the Rack middleware
       @ws = env['rack.hijack'].call
       # Set up the socket client for the proxy
-      @sock = TCPSocket.open(@console.host_name, @console.port)
+      Timeout.timeout(30) { @sock = TCPSocket.open(@console.host_name, @console.port) }
```

Comment:
Is it possible to monkey patch and add `TCPSocket#open_nonblocking`

ref: http://www.mikeperham.com/2009/03/15/socket-timeouts-in-ruby/
ref: https://spin.atomicobject.com/2013/09/30/socket-connection-timeout-ruby/

Seems this should have gotten into ruby by now.